src/main.js

```diff
@@ -1,282 +1,238 @@
-import addDays from 'date-fns/add_days';
-import addMonths from 'date-fns/add_months';
-import endOfYear from 'date-fns/end_of_year';
-import differenceInDays from 'date-fns/difference_in_days';
+import isSameDay from 'date-fns/is_same_day';
 import getDate from 'date-fns/get_date';
 import getDayOfYear from 'date-fns/get_day_of_year';
-import getMonth from 'date-fns/get_month';
-import getYear from 'date-fns/get_year';
-import lastDayOfYear from 'date-fns/last_day_of_year';
-import setDate from 'date-fns/set_date';
-import setDay from 'date-fns/set_day';
-import setDayOfYear from 'date-fns/set_day_of_year';
-import setMonth from 'date-fns/set_month';
 import subDays from 'date-fns/sub_days';
-import subMonths from 'date-fns/sub_months';
-import subWeeks from 'date-fns/sub_weeks';
 import subYears from 'date-fns/sub_years';
-import { applyOffset, formatDate, getSubtractionFn } from './utils';
+import getDaysInYear from 'date-fns/get_days_in_year';
+import getDaysInMonth from 'date-fns/get_days_in_month';
+import isMonday from 'date-fns/is_monday';
+import isSunday from 'date-fns/is_sunday';
+import { applyOffset, formatDate, getSubtractionFn, getStartOfFn, getEndOfFn, getDiffFn } from './utils';
 
-// :: String -> Option(String) -> Object
-function retrievePredefindedDateRange(key, base = Date()) {
-  if (key === 'today') {
-    const date = formatDate(base);
-    return { start: date, end: date };
-  }
-
-  if (key === 'yesterday') {
-    const yesterday = formatDate(subDays(base, 1));
-    return { start: yesterday, end: yesterday };
-  }
-
-  if (key === 'year_to_date') {
-    return {
-      start: formatDate(setDayOfYear(base, 1)),
-      end: formatDate(base),
-    };
-  }
-
-  if (key === 'all_time') {
-    return {
-      start: formatDate(subYears(base, 3)),
-      end: formatDate(base),
-    };
-  }
-
-  throw new Error(`Unrecognized date range: ${key}`);
-}
+const formatRange = (start, end) => ({
```

Comment:
Let's try and stay consistent with the type signatures

Reply:
I don't really get what is that format, I'm used to jsdoc what is that convention?

Reply:
It's following Haskell convention, with a bit of my own taste to adapt it to JavaScript.
I've always found JSDoc extremely verbose and bloated. You're free to change it if you feel strongly about it, but let's stay consistent. Also in this case, this would be written like this: `(Date | Int | String) -> (Date | Int | String) -> Object`

Reply:
No that's fine, I'm not a big fan of Jsdoc. Would love to add flowjs